nixos/modules/tasks/network-interfaces.nix

```diff
@@ -54,6 +54,25 @@ let
   subsystemDevice = interface:
     "sys-subsystem-net-devices-${escapeSystemdPath interface}.device";
 
+  # From https://www.freedesktop.org/software/systemd/man/machine-id.html:
+  # "[The machine] ID uniquely identifies the host.
+  # It should be considered "confidential", and must not be exposed
+  # in untrusted environments, in particular on the network.
```

Comment:
The same reasoning still applies as in the other PR. The machine ID shouldn't be part of your nixos configuration, if you intend to keep it stable across reboots, use the hooks provided in the impermanence setup to cause this file from being preserved across reboots.

Reply:
I don't understand the reasoning. Machineid is static across reboots if you don't have a impermanence setup, and it's a world readable file anyways, so putting it in to nix store isn't a security risk. 

Instead a note could be added, that this shouldn't be used if the nix store is exposed over the network. 

Also, generally we (NixOS) try to declare our systems as much as possible in our configuration files, and machineid is not really any exception. And having hostId derived form the machineId seems like a good thing anyways according to the docs.

Reply:
(I guess what I'm saying is, it's confidential to the network, not to the host itself, given the uog=r permissions)

Reply:
I agree with @eyJhb above. We can add a warning to the option documentation but I don't see it as a reason to not add this NixOS option.

Reply:
Hmmh, if this is added, there should definitely be a lot more context for the specific usecase for this. Also, `networking.` seems to be the wrong namespace.

Also, is `environment.etc` not sufficient to describe this?